include/L.h

```diff
@@ -561,10 +561,10 @@ class L_function
 
     //#include "Ldirichlet_series.h" //for computing Dirichlet series
     Complex partial_dirichlet_series(Complex s, long long N1, long long N2);
-    Complex dirichlet_series(Complex s, long long N);
+    Complex dirichlet_series(Complex s, long long N=-1);
 
-    Complex partial_dirichlet_series_via_blfi(Complex s, long long N1, long long N, long long K, Double epsilon);
-    Complex dirichlet_series_via_blfi(Complex s, long long N, Double blfi_interval_length, Double epsilon);
+    Complex partial_dirichlet_series_via_blfi(Complex s, long long N1=1, long long N=-1, long long K=1, Double epsilon=1.e-14);
```

Comment:
Where do you get the default values for parameters here (and other places in this header file)? It's not clear. It's also not clear to me  why they are needed.

Reply:
They are only moved from the templates in `include/Ldirichlet_series.h`. If I move the one in `dirichlet_series` back clang++ will fail with:
```
../include/Ldirichlet_series.h:53:39: error: default arguments cannot be added
      toIn file included from  anLriemannsiegel.cc out-of-line: definition24 :
ofIn file included from  ../include/L.ha: 612member:
 of a../include/Ldirichlet_series.h :class53 :template39:
 error: default arguments cannot be added
      to an out-of-line definition of a dirichlet_series(Complex s, long long N=-1)member
 of                                      ^ ~~ 
a class template
dirichlet_series(Complex s, long long N=-1)
```
Reason is they should not be on the definition but on the declaration. See the second answer in https://stackoverflow.com/questions/25647199/c-default-arguments-cannot-be-added-to-an-out-of-line-definition-of-a-member for example.